### PR TITLE
Correctly implement autocomplete early return

### PR DESCRIPTION
**Pull request by uranusjr**
Found this bug while working on #15525. This is properly tested now.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/www/views.py [hunk 1]
```diff
@@ -4162,7 +4162,7 @@ def class_permission_name(self):
         """Returns appropriate permission name depending on request method name."""
         if request:
             action_name = request.view_args.get("name")
-            _, method_name = request.url_rule.endpoint.split(".")
+            _, method_name = request.url_rule.endpoint.rsplit(".", 1)
```
Comment (uranusjr) [inline airflow/www/views.py:4165]: Oops, this is unrelated and should have been submitted separately. A trivial cleanup though so probably OK to have it here.

**Review by ashb — APPROVED** [review]
